### PR TITLE
feat: cvc nickname gpay

### PR DESCRIPTION
**Pull request by ArushKapoorJuspay**
Following features and fixes have been added - 

- added check to render cvv for saved card methods only when requires_cvv

https://github.com/juspay/hyperswitch-web/assets/121166031/2f6ee535-b109-474d-803a-4009b975ac2a

- added support for nickname when passing customerAcceptance

https://github.com/juspay/hyperswitch-web/assets/121166031/53895aa5-d6fb-4a91-8688-f22ae20e8952

- switched GPay and ApplePay to oneClickButtons in case required fields are prefilled

https://github.com/juspay/hyperswitch-web/assets/121166031/d260cc48-6732-438d-a908-b371de3fdd53

### CODE REVIEW

File: src/Utilities/DynamicFieldsUtils.res [hunk 1]
```diff
@@ -695,3 +695,24 @@ let useSubmitCallback = () => {
...
+      ~paymentMethod,
+      ~paymentMethodType=PaymentUtils.getPaymentMethodName(
+        ~paymentMethodType=paymentMethod,
+        ~paymentMethodName=paymentMethodType,
+      ),
+    )->Belt.Option.getWithDefault(PaymentMethodsRecord.defaultPaymentMethodType)
+  }, (list, paymentMethod, paymentMethodType))
+}
+
+let areAllRequiredFieldsPrefilled = (~list, ~paymentMethod, ~paymentMethodType) => {
```
Comment (PritishBudhiraja) [inline src/Utilities/DynamicFieldsUtils.res:712]: Hook inside a function call. Pleas check once

File: src/Utilities/PaymentBody.res [hunk 2]
```diff
@@ -1,29 +1,38 @@
...
+    ("card_number", cardNumber->CardUtils.clearSpaces->Js.Json.string),
+    ("card_exp_month", month->Js.Json.string),
+    ("card_exp_year", year->Js.Json.string),
+    ("card_holder_name", cardHolderName->Js.Json.string),
+    ("card_cvc", cvcNumber->Js.Json.string),
+    ("card_issuer", ""->Js.Json.string),
+  ]
+
+  if nickname != "" {
+    cardBody->Js.Array2.push(("nickname", nickname->Js.Json.string))->ignore
```
Comment (PritishBudhiraja) [inline src/Utilities/PaymentBody.res:23]: Above it was nick_name & here nickname please confirm

**Review by seekshiva — APPROVED** [review]

**Review by seekshiva — CHANGES_REQUESTED** [review]
areAllRequiredFieldsPrefilled should be a hook

File: src/Utilities/DynamicFieldsUtils.res (hunk 1 above)
Comment (PritishBudhiraja) [reply src/Utilities/DynamicFieldsUtils.res:712]: Addressed

**Review by seekshiva — APPROVED** [review]

**Review by PritishBudhiraja — APPROVED** [review]

File: src/Utilities/PaymentBody.res (hunk 2 above)
Comment (PritishBudhiraja) [reply src/Utilities/PaymentBody.res]: Updated

**Review by akash-kamble-juspay — DISMISSED** [review]

**Pull request merged**